packages/nocodb/src/lib/meta/helpers/webhookHelpers.ts

```diff
@@ -72,6 +72,26 @@ export async function validateCondition(filters: Filter[], data: any) {
       case 'notnull':
         res = data[field] !== null;
         break;
+      case 'allof':
+        res = (filter.value?.split(',') ?? []).every((item) =>
+          (data[field]?.split(',') ?? []).includes(item)
+        );
+        break;
+      case 'anyof':
+        res = (filter.value?.split(',') ?? []).some((item) =>
+          (data[field]?.split(',') ?? []).includes(item)
+        );
+        break;
+      case 'nallof':
+        res = !(filter.value?.split(',') ?? []).every((item) =>
+          (data[field]?.split(',') ?? []).includes(item)
+        );
```

Comment:
`notallof` - I guess it should check all of them are not matching, right now it will be true even if one is not matching.

```
  res = (filter.value?.split(',') ?? []).every((item) =>
          !(data[field]?.split(',') ?? []).includes(item)
        );

```

The same applies for `nanyof`

```
res = (filter.value?.split(',') ?? []).some((item) =>
          !(data[field]?.split(',') ?? []).includes(item)
        );
```

Correct me if I'm wrong

Reply:
Let us take a look at "does not contain any of" (`nanyof`) first. Consider the filter `nanyof Foo,Bar` and two tag values:

1. Foo,Bar,Quux
2. Foo,Quux

Both 1, 2 should *not* be shown, because they contain some of the items in the filter.

Let us then take a look at the implementation.

```ts
      case 'nanyof':
        res = !(filter.value?.split(',').map((item) => item.trim()) ?? []).some((item) =>
          (data[field]?.split(',') ?? []).includes(item)
        );
        break;
```

In this case, `(filter.value?.split(',').map((item) => item.trim()) ?? [])` evaluates to `["Foo", "Bar"]` and so the `some` evaluates to `true` in both case. Which means that `res` is false, and the rows will not be shown.

So I *think* the implementation of `nanyof` is correct, but I am not 100% sure. What do you think?

Reply:
Now it makes sense :+1: , you can update the docs part as well to make it clear - https://github.com/nocodb/nocodb/blob/develop/packages/noco-docs/content/en/developer-resources/rest-apis.md#comparison-operators